### PR TITLE
remember email when setting expiration date

### LINKED ISSUE #22947: Share link: user mail is disappearing after selecting expiration date
### Steps to reproduce
1. Enable: Allow users to share via link > Allow users to send mail notification for shared files
2. Share link
3. Input: Email link to person
4. Select expiration date with date selector

### Expected behaviour
- Email address shouldn't disappear after setting an expiration data

### Actual behaviour
- Email address disappears after setting an expiration data

### Server configuration
**ownCloud version:** (see ownCloud admin page)
8.2.2, 8.2.3RC2

**Updated from an older ownCloud or fresh install:**
fresh install

### Client configuration
**Browser:**
Chrome 49

**Operating system:**
Mac OS X 10.11.3

@MorrisJobke Who can help here?

00004983

### PR DESCRIPTION
**Pull request by ChristophWurst**
fixes #22947

### CODE REVIEW

**Comment by ChristophWurst** [comment]
@jancborchardt @butonic if the order is changed so the email field is shown as last element [here](https://github.com/owncloud/core/blob/master/core/js/sharedialogview.js#L26-L29), the expiration checkbox is shown above the 'share link' checkbox:
bildschirmfoto von 2016-04-14 08-59-46

That makes the usability even worse, because you don't have the option to set an expiration date if it's not a link share…
I guess we have to restructure those views and make the expiration stuff a sub-view of the link sharing view.

**Comment by jancborchardt** [comment]
> I guess we have to restructure those views and make the expiration stuff a sub-view of the link sharing view.

For sure … the order should be:

- Share with others
- Share link
  - Password protect
  - Set expiration date
  - Email link

**Comment by ChristophWurst** [comment]
@jancborchardt this is what it looks like now (I updated the commit):
bildschirmfoto von 2016-04-14 16-17-24

I'm not sure if it's clear what the different input fields are used for. Should we add some separation between the share information and the email form?

**Comment by jancborchardt** [comment]
Maybe it should be a button »Send link via email« by default. On clicking, the input field appears, with an icon-confirm button on the right (like the integrated button on the log in page, not a separate line). What do you think?

**Comment by PVince81** [comment]
Let's make the label say "Send link via email" for now for a backportable solution.

File: core/js/sharedialogmailview.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			}
+			
+			var isLinkShare = this.model.get('linkShare').isLinkShare;
+
+			this.$el.html(linkShareTemplate({
+				cid: this.cid,
+				shareAllowed: true,
+				mailPublicNotificationEnabled: isLinkShare && this.configModel.isMailPublicNotificationEnabled(),
+				mailPrivatePlaceholder: t('core', 'Email link to person'),
+				mailButtonText: t('core', 'Send link via email'),
```
Comment (ChristophWurst) [inline core/js/sharedialogmailview.js]: @PVince81 It already says 'Send link via email' ;-)

Comment (PVince81) [inline core/js/sharedialogmailview.js]: Ah nice... was looking at the old screenshot

**Comment by PVince81** [comment]
Tested, works :+1: 

@karlitschek backport for v9.0.2 ? (blue)

**Comment by PVince81** [comment]
@karlitschek also v8.2.4 apparently

**Comment by PVince81** [comment]
Needs a second reviewer @rullzer @jancborchardt

**Comment by karlitschek** [comment]
great. please backport :+1

**Comment by jancborchardt** [comment]
Nice

**Comment by ChristophWurst** [comment]
Backport 8.2: https://github.com/owncloud/core/pull/24152
Backport 9: https://github.com/owncloud/core/pull/24153

Note that I backported the simple fix for the actual issue, not the refactoring of the views which might break stuff.

**Pull request merged**